specs/agents/tracing-instrumentation-db.md

```diff
@@ -88,6 +88,9 @@ The following fields are relevant for database and datastore spans. Where possib
 | __**service.target._**__ |<hr/>|<hr/>|
 |`_.type`| `s3` ||
 |`_.name`| e.g. `my-bucket`, `accesspoint/myendpointslashes`, or `accesspoint:myendpointcolons` | The bucket name, if available. The s3 API allows either the bucket name or an Access Point to be provided when referring to a bucket. Access Points can use either slashes or colons. When an Access Point is provided, the access point name preceded by accesspoint/ or accesspoint: should be extracted. For example, given an Access Point such as `arn:aws:s3:us-west-2:123456789012:accesspoint/myendpointslashes`, the agent extracts `accesspoint/myendpointslashes`. Given an Access Point such as `arn:aws:s3:us-west-2:123456789012:accesspoint:myendpointcolons`, the agent extracts `accesspoint:myendpointcolons`. |
+| __**otel.attributes._**__ |<hr/>|<hr/>|
+|`_["aws.s3.bucket"]`| `my-bucket` | The bucket name, if available. See [OTel Semantic Conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/semantic_conventions/trace/instrumentation/aws-sdk.yml#L435). |
```

Comment:
```suggestion
|`_.aws.s3.bucket`| `my-bucket` | The bucket name, if available. See [OTel Semantic Conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/semantic_conventions/trace/instrumentation/aws-sdk.yml#L435). |
```

Reply:
@AlexanderWert @david-luna Except this isn't what we are actually sending -- or at least stating it as `otel.attributes.aws.s3.bucket` is misleading I think, because `otel.attributes` needs to be a mapping with a string key named `"aws.s3.bucket"`.

Currently we are sending span object looking like:

```json
{
    "span": {
        "name": "S3 GetObject elasticapmtest-bucket-20230426153626",
        "type": "storage",
        "id": "912aad817e62dbc6",
        // ...
        "otel": {
            "attributes": {
                "aws.s3.bucket": "elasticapmtest-bucket-20230426153626",
                "aws.s3.key": "aDir/aFile.txt-does-not-exist"
            }
        }
    }
}
```

That results in a document with:

```json
{
  "_index": ".ds-traces-apm-default-2023.04.06-000002",
  "_source": {
    "labels": {
      "aws_s3_key": "aDir/aFile.txt",
      "aws_s3_bucket": "elasticapmtest-bucket-20230426153920"
    },
    // ...
  },
  "fields": {
    "span.name": [
      "S3 GetObject elasticapmtest-bucket-20230426153920"
    ],
    "labels.aws_s3_key": [
      "aDir/aFile.txt"
    ],
    "labels.aws_s3_bucket": [
      "elasticapmtest-bucket-20230426153920"
    ],
    // ...
  }
}
```

If we instead send:

```json
{
    "span": {
        "name": "S3 GetObject elasticapmtest-bucket-20230426155211",
        // ...
        "otel": {
            "attributes": {
                "aws": {
                    "s3": {
                        "bucket": "elasticapmtest-bucket-20230426155211",
                        "key": "aDir/aFile.txt-does-not-exist"
                    }
                }
            }
        }
    }
}
```

The resulting document does not include labels. I believe this code drops
a mapping value that is an object: https://github.com/elastic/apm-data/blob/d1edc48bb5e38a58921a82c2c141c89d93eb0e6b/input/elasticapm/internal/modeldecoder/v2/decoder.go#L1509-L1551

Reply:
Ou, I see. That's my fault then. To make it explicit, let's add then a note in the spec to make clear it's a string and not an object

Reply:
https://github.com/elastic/apm/pull/789 to clarify